corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js

```diff
@@ -209,6 +210,25 @@ FormplayerFrontend.on("start", function (options) {
     }
 });
 
+FormplayerFrontend.reqres.setHandler('getCurrentAppId', function() {
+    // First attempt to grab app id from URL
+    var urlObject = Util.currentUrlToObject(),
+        user = FormplayerFrontend.request('currentUser'),
+        appId;
+
+    appId = urlObject.appId;
+
+    if (appId) {
+        return appId;
+    }
+
+    // If it's not in the URL, then we are either on the home screen of formplayer
+    // and there is no app selected, or we are in preview mode.
+    appId = user.previewAppId;
```

Comment:
so i actually looked into doing that. and it doesn't quite work, because when you set the appId in the URL, the fragment changes from `#single_app/<app_id>` to `#{...json..}` which means when you refresh or go back, you go to the formplayer home screen instead of the single app view. we mit in the future by doing: `#single_app/<app_id>/{...json...}` but seems kind of hacky. another approach would be to drop the `single_app` route and just use it as a parameter in the json blob